### ISSUE TITLE
Tree not updating correct after last update

### ISSUE THREAD

**Issue by PerThomsen** [samuelet/indexmenu#302]
After the last update the tree dos not reflect correct.  New pages do not appear and deleted pages still remain. When I put &purge=1 at the end of the URL it displays correctly

**Comment by Klap-in** [comment]
More people have now reported this issue, see also #296. So far I could not yet reproduce it on my own setup. But as the reports keep coming, there happens somewhere a wrong handling/triggering of the cache.

**Comment by IcEPiQQ** [comment]
had the same issue!
I solved that problem by 
- changing the defaultoptions setting in the Configuration Manager to "treenew" and 
- changing the plugin»indexmenu»aclcache to "user"

**Comment by PerThomsen** [comment]
> had the same issue! I solved that problem by
> 
>     * changing the defaultoptions setting in the Configuration Manager to "treenew" and
> 
>     * changing the plugin»indexmenu»aclcache to "user"

When I do that, my action-menu (right-click on menu) disappears ;)

**Comment by Klap-in** [comment]
`treenew` has no contextmenu (yet..). (Btw, I doubt if I should readd it, see #244 

All the reports give different small clues. If I can find time, my intent is to test multiple combinations of caching in the hope to find the cause. Unfortunately, so far I have not yet reproduced it.

**Comment by e-dschungel** [comment]
Hi,

I got the same or a similar issue. New sub pages are not shown in the indexmenu of the top page. If I edit the top page and press preview the new page is there. If I edit the page (just add a newline) and save the new page is in the index. So I guess that's an issue with the cache.

Used syntax (top page):
`{{indexmenu>.:Notizen#0|group}} `
Subpages are created in in the Notizen Namespace.

Release 2024-02-06a "Kaos", PHP 8.2.16
indexmenu defaultoptions: tsort nsort nogroup
aclcache: groups
Installed Plugins
-     Authad
-     Authldap
-     Authpdo
-     Commentsyntax
-     Folded
-     Indexmenu
-     Move
-     Pagelist
-     Tag
-     Wrap

If more data is helpful, please let me know.

**Comment by Stikus** [comment]
Yes, I have same problem (wrote here: https://github.com/samuelet/indexmenu/issues/296#issuecomment-1943647099) - I can provide any required data, and we need this fix.

**Comment by hurik** [comment]
> had the same issue! I solved that problem by
> 
>     * changing the defaultoptions setting in the Configuration Manager to "treenew" and
> 
>     * changing the plugin»indexmenu»aclcache to "user"

I have the same problem, and the above solution doesn't help. On my startpage i have a menu, It shows the same as the sidebar (for mobile users). When I log out, the sidebar is correct, but the startpage shows everything, like I'm still logged in.

**Comment by Stikus** [comment]
@Klap-in Hello, are you still sick? Any ETA for fix?

**Comment by Klap-in** [comment]
I'm recovered. But I´m not able to provide a ETA because I'm not able to reproduce the situation, therefore, difficult get grip on it. Further, I've very limited time due to care for a newborn.

**Comment by f1ynng8** [comment]
Yes, I have the exactly same issue.

When using the default js tree (tree old), the indexemnu will not handle new page or deleted page, you need to pure the page to make indexmenu  display it.

The solution is to use treenew either in defaultconfigs or in the indexmenu code, like ` {{indexmenu>..#1|js#thread navbar treenew skipfile=/sidebar/}} `

However I encounter another problem, after changing to treenew, the indexmenu only use the default theme and I can't change them by #theme.

**Comment by f1ynng8** [comment]
btw, there is a simple workaround if you using `treeold`, just add `~~NOCACHE~~ ` the your sidebar page, like this:
```
~~NOCACHE~~ 
{{indexmenu>..#1|js#leopard.png navbar treeold skipfile=/sidebar/}} 
```

I am using this workaround, because I like leopard.png theme but indexmenu can't change theme while using treenew https://github.com/samuelet/indexmenu/issues/313

**Comment by Stikus** [comment]
@f1ynng8 Thx for advice, `~~NOCACHE~~`  helped us. But we're still waiting for proper solution.

**Comment by Klap-in** [comment]
Nope, the very strange thing here is that there are submitted several bit different reports,that seems kind of related to caching, but which I can not reproduce for myself.

This makes it difficult for me to come up with a solution. Also I have limited time, recently tried another unsuccessful attempt to reproduce, but I do not know how to speed up a remedy.

**Comment by e-dschungel** [comment]
@Klap-in: Would access to an affected installation help? If yes, I could provide one.

@PerThomsen, @vincent1890, @Stikus, @f1ynng8, @hurik, @IcEPiQQ: My affected installation is using a [Symlink Farm](https://www.dokuwiki.org/tips:symlink_farm). Are you using any kind of wiki farm as well? Maybe that's what's causes this error.
